cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java

```diff
@@ -285,4 +239,13 @@ public long getEnd() {
       return end;
     }
   }
+
+  private Constants.FieldLineage.Direction parseDirection(String directionStr) throws BadRequestException {
+    try {
+      return Constants.FieldLineage.Direction.valueOf(directionStr);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException("Direction must be specified to get the field lineage " +
+              "summary and should be one of the 'incoming', 'outgoing', or 'both'.");
```

Comment:
better to generate the list of values from the enum, in case the enum gets changed in the future.
Also, use standard indentation.

Reply:
fixed.